src/runtime/composables/search.ts

```diff
@@ -0,0 +1,113 @@
+import MiniSearch, { type Options as MiniSearchOptions } from 'minisearch'
+import type { UseFetchOptions } from 'nuxt/app'
+import { useRuntimeConfig, useFetch } from '#imports'
+
+export const defineMiniSearchOptions = <DataItem>(options: MiniSearchOptions<DataItem>) => {
+  return ref(options)
+}
+
+export const searchContent = async <DataItem>(search: MaybeRefOrGetter<string>, options: { miniSearch?: MaybeRefOrGetter<MiniSearchOptions<DataItem>>, fetch?: MaybeRefOrGetter<UseFetchOptions<string | DataItem[]>> } = {}) => {
```

Comment:
What is the use case of `options.fetch`? 
I don't think this is good idea to accept fetch options here. Calling this fetch is more like an internal stuff here.